src/linkplay/utils.py

```diff
@@ -67,6 +70,46 @@ async def session_call_api_ok(
         raise LinkPlayRequestException(f"Didn't receive expected OK from {endpoint}")
 
 
+async def call_tcpuart(
+    reader: asyncio.StreamReader, writer: asyncio.StreamWriter, cmd: str
+) -> str:
+    """Get the latest data from TCP UART service."""
+    payload_header = "18 96 18 20 "
+    payload_length = format(len(cmd), "02x")
+    payload_command_header = " 00 00 00 c1 02 00 00 00 00 00 00 00 00 00 00 "
+    payload_command_content = " ".join(hex(ord(c))[2:] for c in cmd)
+    data = None
+
+    async with async_timeout.timeout(API_TIMEOUT):
+        writer.write(
+            bytes.fromhex(
+                payload_header
+                + payload_length
+                + payload_command_header
+                + payload_command_content
+            )
+        )
+        data = str(repr(await reader.read(1024))).encode().decode("unicode-escape")
+
+    if not data:
+        raise LinkPlayRequestException("No data received from socket")
+
+    return data
```

Comment:
```suggestion
    async with async_timeout.timeout(API_TIMEOUT):
        writer.write(
            bytes.fromhex(
                payload_header
                + payload_length
                + payload_command_header
                + payload_command_content
            )
        )
        
        data: bytes = await reader.read(1024)

        if data == b'':
            raise LinkPlayRequestException("No data received from socket")

        return str(repr(data)).encode().decode("unicode-escape")
```

Reply:
Upon looking at it a bit closer, it seems `StreamReader` will return `b''` when there's no data. I also think the `return str(repr(data)).encode().decode("unicode-escape")` can be optimized without having to encode and decode it?